From 8fd6dbd941a245013d970667851f6f96e00e690f Mon Sep 17 00:00:00 2001
From: Howard Hinnant <hhinnant@apple.com>
Date: Sun, 25 Aug 2013 17:00:57 +0000
Subject: [PATCH 1/2] Propagate the removal of _LIBCPP_CANTTHROW (r189046) to
 libcxxabi.  This fixes http://llvm.org/bugs/show_bug.cgi?id=16996.

git-svn-id: https://llvm.org/svn/llvm-project/libcxxabi/trunk@189194 91177308-0d34-0410-b5e6-96231b3b80d8
(cherry picked from commit cdf7d557bd6835f66fe707ca7b1a8f22010ffb66)
---
 src/stdexcept.cpp | 10 +++++-----
 1 file changed, 5 insertions(+), 5 deletions(-)

diff --git a/src/stdexcept.cpp b/src/stdexcept.cpp
index de859db..460ff8b 100644
--- a/src/stdexcept.cpp
+++ b/src/stdexcept.cpp
@@ -66,9 +66,9 @@ private:
 
 public:
     explicit __libcpp_nmstr(const char* msg);
-    __libcpp_nmstr(const __libcpp_nmstr& s) _LIBCPP_CANTTHROW;
-    __libcpp_nmstr& operator=(const __libcpp_nmstr& s) _LIBCPP_CANTTHROW;
-    ~__libcpp_nmstr() _LIBCPP_CANTTHROW;
+    __libcpp_nmstr(const __libcpp_nmstr& s) _NOEXCEPT;
+    __libcpp_nmstr& operator=(const __libcpp_nmstr& s) _NOEXCEPT;
+    ~__libcpp_nmstr();
     const char* c_str() const _NOEXCEPT {return str_;}
 };
 
@@ -84,7 +84,7 @@ __libcpp_nmstr::__libcpp_nmstr(const char* msg)
 }
 
 inline
-__libcpp_nmstr::__libcpp_nmstr(const __libcpp_nmstr& s)
+__libcpp_nmstr::__libcpp_nmstr(const __libcpp_nmstr& s) _NOEXCEPT
     : str_(s.str_)
 {
 #if __APPLE__
@@ -94,7 +94,7 @@ __libcpp_nmstr::__libcpp_nmstr(const __libcpp_nmstr& s)
 }
 
 __libcpp_nmstr&
-__libcpp_nmstr::operator=(const __libcpp_nmstr& s)
+__libcpp_nmstr::operator=(const __libcpp_nmstr& s) _NOEXCEPT
 {
     const char* p = str_;
     str_ = s.str_;
-- 
2.2.1

